### ISSUE TITLE
[BUG, WEBSOCKETS] Messages duplication

### ISSUE THREAD

**Issue by SerafimArts** [spiral/roadrunner#733]
I tried this code:

```php
RPC::create('tcp://127.0.0.1:6001')
    ->call('websockets.Publish', new Request([
        'messages' => [new Message([
            'broker' => 'memory',
            'topics' => ['topic-1', 'topic-2'],
            'payload' => 'message',
        ])]
    ]);
```

With follow config:
```yaml
rpc:
    listen: tcp://127.0.0.1:6001

server:
    command: "php worker.php"
    relay: "pipes"

http:
    address: 127.0.0.1:80
    middleware: [ "websockets" ]
    pool:
        num_workers: 2

websockets:
    pubsubs: [ "memory" ]
    path: "/ws"
```

I expected to see this happen: 
```json
{"topic":"topic-1","payload":"message"}
{"topic":"topic-2","payload":"message"}
```

Instead, this happened:
```json
{"topic":"topic-1","payload":"message"}
{"topic":"topic-2","payload":"message"}
{"topic":"topic-1","payload":"message"}
{"topic":"topic-2","payload":"message"}
```

The version of RR used: `rr.exe version 2.3.1-beta.3 (build time: 2021-06-16T13:34:27+0000, go1.16.4)`

----
- The error does not depend on the number of workers in `num_workers` config section.
- The error changes its behavior if you subscribe to a different number of topics. For example, this is how the result looks for 3 channels:
```js
const ws = new WebSocket('ws://127.0.0.1/broadcast');

ws.onopen = e => {
    const message = {
        command: 'join',
        broker:  'memory',
        topics:  ['topic-1', 'topic-2', 'topic-3']
    };

    ws.send(JSON.stringify(message));
};

ws.onmessage = e => {
    const message = JSON.parse(e.data);
}
```

Then send only 1 message with payload "message" into 3 topics:

![image](https://user-images.githubusercontent.com/2461257/122457137-01bc1400-cfb7-11eb-9d16-54e8d72780d7.png)

**Comment by wolfy-j** [comment]
I've seen this error during 1st implementation. It was solved by the Client. Router abstraction to ensure only unique subscriptions.

**Comment by rustatian** [comment]
I can not reproduce this issue on the latest build.
Connecting to the 4 topics:
![image](https://user-images.githubusercontent.com/8040338/122473326-60d75400-cfca-11eb-93dd-9ba8b014d0e0.png)  

----
Connected:
![image](https://user-images.githubusercontent.com/8040338/122473556-b1e74800-cfca-11eb-8969-e837cdad1e6d.png)  

----
Sent 1 message to the 3 topics within initial 4:
`makeMessage("join", "memory", []byte(data), "foo", "foo2", "foo3")`

----
Received w/o duplicates:
![image](https://user-images.githubusercontent.com/8040338/122473827-07bbf000-cfcb-11eb-89d1-a72c0f5ef276.png)

----
Left 1 topic (foo), received 2 messages w/o duplicates:
![image](https://user-images.githubusercontent.com/8040338/122474031-4ea9e580-cfcb-11eb-8ba3-f750749fb201.png)

----

Rejoin, sent message, no duplicates:

![image](https://user-images.githubusercontent.com/8040338/122474145-7b5dfd00-cfcb-11eb-9dc0-1032be5f6472.png)

**Comment by rustatian** [comment]
fixed in the #732, will be closed automatically after PR got merged.
